drivers/usb/udc/udc_ambiq.c

```diff
@@ -0,0 +1,915 @@
+#define EP_HS_MPS 512U
+
+/* UDC driver private data structure definition */
+struct udc_ambiq_data {
+	struct k_thread thread_data;
+	void *usbHandle;
+	am_hal_usb_dev_speed_e eUsbSpeed;
+	uint8_t setup[8];
+};
+
+/* UDC driver configuration structure definition */
+struct udc_ambiq_config {
+	uint32_t num_endpoints;
+	int speed_idx;
+	struct udc_ep_config *ep_cfg_in;
+	struct udc_ep_config *ep_cfg_out;
+	void (*make_thread)(const struct device *dev);
+	void (*irq_enable_func)(const struct device *dev);
+	void (*irq_disable_func)(const struct device *dev);
+};
+
+/* Static variables */
+const static struct device *udc_ambiq_dev;
+static bool udc_ambiq_ctrl_pending_in_ack;
+static bool udc_ambiq_ctrl_pending_setup;
+static uint8_t udc_ambiq_ctrl_pending_setup_buffer[8] = {0};
+static bool udc_ambiq_ctrl_setup_recv_at_status_in;
+
+/* Static function prototype */
+static int udc_ambiq_rx(const struct device *dev, uint8_t ep, struct net_buf *buf);
+
+/* Feed buffer to Control OUT endpoint */
+static int usbd_ctrl_feed_dout(const struct device *dev, const size_t length)
+{
+	struct udc_ep_config *cfg = udc_get_ep_cfg(dev, USB_CONTROL_EP_OUT);
+	struct net_buf *buf;
+
+	buf = udc_ctrl_alloc(dev, USB_CONTROL_EP_OUT, length);
+	if (buf == NULL) {
+		return -ENOMEM;
+	}
+
+	k_fifo_put(&cfg->fifo, buf);
+	if (length) {
+		udc_ambiq_rx(dev, cfg->addr, buf);
+	}
+
+	return 0;
+}
+
+/* Non-control EP data transmit via USB HAL */
+static int udc_ambiq_tx(const struct device *dev, uint8_t ep, struct net_buf *buf)
+{
+	const struct udc_ambiq_data *priv = udc_get_private(dev);
+	uint32_t status;
+
+	if (udc_ep_is_busy(dev, ep)) {
+		LOG_WRN("ep 0x%02x is busy!", ep);
+		return 0;
+	}
+	udc_ep_set_busy(dev, ep, true);
+
+	status = am_hal_usb_ep_xfer(priv->usbHandle, ep, buf->data, buf->len);
+	if (status != AM_HAL_STATUS_SUCCESS) {
+		udc_ep_set_busy(dev, ep, false);
+		LOG_ERR("am_hal_usb_ep_xfer write failed(0x%02x), %d", ep, (int)status);
+		return -EIO;
+	}
+
+	return 0;
+}
+
+/* Non-control EP data receive via USB HAL */
+static int udc_ambiq_rx(const struct device *dev, uint8_t ep, struct net_buf *buf)
+{
+	struct udc_ambiq_data *priv = udc_get_private(dev);
+	struct udc_ep_config *cfg = udc_get_ep_cfg(dev, USB_CONTROL_EP_OUT);
+	uint32_t status;
+	uint16_t rx_size = buf->size;
+
+	if (udc_ep_is_busy(dev, ep)) {
+		LOG_WRN("ep 0x%02x is busy!", ep);
+		return 0;
+	}
+	udc_ep_set_busy(dev, ep, true);
+
+	/* Make sure that OUT transaction size triggered doesn't exceed EP's MPS */
+	if ((ep != USB_CONTROL_EP_OUT) && (cfg->mps < rx_size)) {
+		rx_size = cfg->mps;
+	}
+
+	status = am_hal_usb_ep_xfer(priv->usbHandle, ep, buf->data, rx_size);
+	if (status != AM_HAL_STATUS_SUCCESS) {
+		udc_ep_set_busy(dev, ep, false);
+		LOG_ERR("am_hal_usb_ep_xfer read(rx) failed(0x%02x), %d", ep, (int)status);
+		return -EIO;
+	}
+
+	return 0;
+}
+
+/* Handler for USB event callbacks from USB HAL */
+static void udc_evt_callback(am_hal_usb_dev_event_e eDevState)
```

Comment:
no cAmelCase

Reply:
Hi @jfischer-no, I have searched through and removed cAmelCase variable name throughout the file. Thanks for catching this 😉